src/core_write.cpp

```diff
@@ -88,7 +88,12 @@ string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDecode)
         }
         if (0 <= opcode && opcode <= OP_PUSHDATA4) {
             if (vch.size() <= static_cast<vector<unsigned char>::size_type>(4)) {
-                str += strprintf("%d", CScriptNum(vch, false).getint());
+                if ( (lastopcode == OP_NAME_NEW || lastopcode == OP_NAME_UPDATE || lastopcode == OP_NAME_FIRSTUPDATE )
+                    && vch.size() > static_cast<vector<unsigned char>::size_type>(0) ) {
```

Comment:
I completely agree. Would you like me to correct this and resubmit?

Any thoughts on the decode failure in the event bare 0x00 is encountered in the script? It seems to me that the integer print in the event of a so-called 0-length or empty vch is almost an accidental oversight which just happens to work.